src/consensus/pbft/libbyz/replica.cpp

```diff
@@ -479,6 +479,57 @@ void Replica::playback_request(ccf::Store::Tx& tx)
 
   exec_command(vec_exec_cmds, playback_byz_info, 1);
   did_exec_gov_req = did_exec_gov_req || playback_byz_info.did_exec_gov_req;
+  brt.add_request(req.release());
+}
+
+void Replica::populate_certificates(Pre_prepare* pp, bool add_mine)
+{
+  if (pp->seqno() <= 0)
+  {
+    // first pre prepare will not contain proofs for a previous pre prepare
+    return;
+  }
+
+  auto principals_with_proofs = pp->get_prev_pp_valid_principals();
+  auto prev_seqno = pp->seqno() - 1;
+  if (!plog.within_range(prev_seqno))
+  {
+    LOG_DEBUG_FMT(
+      "seqno {} is out of range, can not add prepare proofs to plog",
+      prev_seqno);
+    return;
+  }
+  auto& prev_prepared_cert = plog.fetch(prev_seqno);
```

Comment:
move this inside the scope of  "if (prev_pp != 0)"

Reply:
but we use `prev_prepared_cert` to get `prev_pp` and then check if it is `0`  or not